be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/RollCallValidator.scala

```diff
@@ -76,10 +75,19 @@ case object RollCallValidator extends MessageDataContentValidator {
       case Some(message: Message) =>
         val data: CloseRollCall = message.decodedData.get.asInstanceOf[CloseRollCall]
 
-        if (!validateTimestampStaleness(data.end)) {
-          Right(validationError(s"stale 'end' timestamp (${data.end})"))
+        val laoId: Hash = rpcMessage.extractLaoId
+        // Note: this id doesnt follow the standard sh256(string, hash, timestamp, string) other events do;
+        // this is why generateValidationId cannot be used
+        val expectedRollCallId: Hash = Hash.fromStrings(
+          EVENT_HASH_PREFIX, laoId.toString, data.closes.toString, data.closed_at.toString
+        )
+
+        if (!validateTimestampStaleness(data.closed_at)) {
+          Right(validationError(s"stale 'closed_at' timestamp (${data.closed_at})"))
         } else if (data.attendees.size != data.attendees.toSet.size) {
           Right(validationError("duplicate attendees keys"))
+        } else if (expectedRollCallId != data.closes) {
+          Right(validationError("unexpected id 'closes'"))
```

Comment:
We're comparing the wrong item here.
the `closes` id should be in the database, but it's `update_id` which should be `== expectedRollCallId`
cf. https://github.com/dedis/student_21_pop/blob/master/fe1-web/model/network/method/message/data/rollCall/CloseRollCall.ts

Reply:
Nice catch! ^^'